inc/Nav_Menus/Component.php

```diff
@@ -59,15 +67,55 @@ public function template_tags() : array {
 		);
 	}
 
+	/**
+	 * Set the navs menus slugs and their description.
+	 *
+	 * @return void
+	 */
+	public function set_navs_menus() {
+		$this->nav_menus = array(
+			'primary' => esc_html__( 'Primary', 'wp-rig' ),
+		);
+	}
+
+	/**
+	 * Get the navs menus slugs and their description.
+	 *
+	 * @return array Associative array of menu location identifiers (like a slug) and descriptive text.
+	 */
+	public function get_navs_menus() : array {
+		return $this->nav_menus;
+	}
+
+	/**
+	 * Get the navs menus slugs.
+	 *
+	 * @return array Array of menu location identifiers (like a slug).
+	 */
+	public function get_menus_slugs() : array {
+		return array_keys( $this->nav_menus );
+	}
+
+	/**
+	 * Get the primary nav menu slug.
+	 *
+	 * @return string Primary menu location identifier (like a slug).
+	 */
+	public function get_primary_menu_slug() : string {
+		$menus_slugs = $this->get_menus_slugs();
+
+		return reset( $menus_slugs );
+	}
+
 	/**
 	 * Registers the navigation menus.
 	 */
 	public function action_register_nav_menus() {
-		register_nav_menus(
-			array(
-				static::PRIMARY_NAV_MENU_SLUG => esc_html__( 'Primary', 'wp-rig' ),
-			)
-		);
+		if ( empty( $this->nav_menus ) ) {
+			return;
+		}
```

Comment:
Same here, I think this should not be checked. If you don't want nav menus at all, it should be encouraged to remove the entire component rather than having it still there without doing anything.